src/runner.py

```diff
@@ -4,27 +4,36 @@
 
 import argparse
 import sys
-from subprocess import run
+from subprocess import run, CalledProcessError
 from time import sleep, time
 
 
 def main() -> None:
-    parser = argparse.ArgumentParser(
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter
-    )
+    parser = argparse.ArgumentParser()
     parser.add_argument(
         "-c",
         "--config",
         default="/conf/bandersnatch.conf",
         help="Configuration location",
     )
-    parser.add_argument("interval", help="Time in seconds between runs")
+    parser.add_argument("interval", help="Time in seconds between runs", type=int)
     args = parser.parse_args()
 
     print(f"Running bandersnatch every {args.interval}s", file=sys.stderr)
     while True:
         start_time = time()
-        run(["/usr/bin/bandersnatch", "--config", args.conf, "mirror"])
+        try:
+            cmd = [
+                sys.executable,
+                "-m",
+                "bandersnatch.main",
+                "--config",
+                args.config,
+                "mirror",
+            ]
+            run(cmd, check=True)
+        except CalledProcessError:
+            sys.exit(1)
```

Comment:
You wouldn't prefer the exception to see the return value from bandersnatch itself? I feel this masks a potential more helpful return code always returning 1 ....

Reply:
Yeah, indeed, that would be better